base/applications/taskmgr/procpage.c

```diff
@@ -931,3 +933,152 @@ int CALLBACK ProcessPageCompareFunc(LPARAM lParam1, LPARAM lParam2, LPARAM lPara
     }
     return ret;
 }
+
+static DWORD _GetModuleFileNameExW(HANDLE hProcess, HMODULE hModule, LPWSTR lpFilename, DWORD nSize)
+{
+    typedef DWORD (WINAPI *PFN_GETMODULEFILENAMEEXW)(HANDLE, HMODULE, LPWSTR, DWORD);
+
+    HMODULE hLibrary;
+    PFN_GETMODULEFILENAMEEXW func;
+    DWORD dwRet = 0;
+
+    hLibrary = LoadLibraryW(L"psapi.dll");
+
+    if (hLibrary)
+    {
+        func = (PFN_GETMODULEFILENAMEEXW)GetProcAddress(hLibrary, "GetModuleFileNameExW");
```

Comment:
Is it possible to use NtQuerySystemInformation() from ntdll here instead maybe?
https://docs.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntquerysysteminformation
Windows XPSP3 taskmgr also uses NtQuerySystemInformation() from ntdll.dll, but does not import from psapi.dll
![image](https://user-images.githubusercontent.com/33393466/151791894-eec48327-d11f-417b-b472-2300c641bca1.png)

Reply:
Hmm, maybe not. Otherwise we would have that information at hand already from other parts of taskmgr.
Guess the current solution to use psapi.dll might be the best then already here. Other peoples ideas are appreciated though.

Reply:
Oh I found something else that may be of (future) interest:
"GetModuleFileNameEx dosen't work with 32-bit app in 64-bit systems, so I've use GetProcessImageFileName with _fullpath and _splitpath_s and its work fine." from
https://stackoverflow.com/questions/30254793/getmodulefilenameex-split-output

Reply:
> Oh I found something else that may be of (future) interest: "GetModuleFileNameEx dosen't work with 32-bit app in 64-bit systems, so I've use GetProcessImageFileName with _fullpath and _splitpath_s and its work fine." from https://stackoverflow.com/questions/30254793/getmodulefilenameex-split-output

Interesting, if it can convert device path to normal path without using `QueryDosDevice` method which is not performance-friendly so it will be a fix for CORE-16145 as well.

Reply:
> without using QueryDosDevice method which is not performance-friendly

It's not as if you were calling that function in a tight loop every fraction of a second...

Reply:
@HBelusca I'll definitely use `GetProcessImageFileName` or `NtQueryInformationProcess` instead of `GetModuleFileNameEx` after finished working on path conversion.

Reply:
The reason we do PRs is for aiming to get things more correct right from the start.

Reply:
I mean portability improvements for shell32 (as suggested by @HBelusca), but for `GetProcessExecutablePath` I'll fix it.

Reply:
`GetProcessExecutablePath` fixed.